tests/test_project.py

```diff
@@ -647,6 +648,80 @@ def test_reuse_info_of_copyright_xor_licensing(empty_directory):
     assert not bar_file_info.spdx_expressions
 
 
+def test_reuse_info_of_copyright_xor_licensing_with_license_dot(
+    empty_directory,
+):
+    """Test a corner case where partial REUSE information is defined inside of a
+    .license file (copyright xor licensing). Get the missing information from
+    the REUSE.toml.
+    """
+    (empty_directory / "REUSE.toml").write_text(
+        cleandoc(
+            """
+            version = 1
+
+            [[annotations]]
+            path = "*.py"
+            SPDX-FileCopyrightText = "2017 Jane Doe"
+            SPDX-License-Identifier = "CC0-1.0"
+
+            [[annotations]]
+            path = "*.py.license"
+            SPDX-FileCopyrightText = "2017 Jane Doe"
+            SPDX-License-Identifier = "MIT"
+            """
+        )
+    )
+    (empty_directory / "foo.py").write_text(
+        cleandoc(
+            """
+            print("This is very strictly copyrighted code")
+            """
+        )
+    )
+    (empty_directory / "foo.py.license").write_text(
+        cleandoc(
+            """
+            SPDX-FileCopyrightText: 2017 John Doe
+            """
+        )
+    )
+    (empty_directory / "bar.py").write_text(
+        cleandoc(
+            """
+            print("This is other very strictly copyrighted code")
+            """
+        )
+    )
+    (empty_directory / "bar.py.license").write_text(
+        cleandoc(
+            """
+            SPDX-License-Identifier: 0BSD
+            """
+        )
+    )
+    project = Project.from_directory(empty_directory)
+
+    foo_infos = project.reuse_info_of("foo.py")
+    assert len(foo_infos) == 2
+    foo_toml_info = [info for info in foo_infos if info.spdx_expressions][0]
+    assert foo_toml_info.source_type == SourceType.REUSE_TOML
+    assert not foo_toml_info.copyright_lines
+    assert "MIT" not in str(foo_toml_info.spdx_expressions)
+    foo_file_info = [info for info in foo_infos if info.copyright_lines][0]
+    assert foo_file_info.source_type == SourceType.DOT_LICENSE
+    assert not foo_file_info.spdx_expressions
```

Comment:
```suggestion
    assert foo_file_info.source_type == SourceType.DOT_LICENSE
    assert not foo_file_info.spdx_expressions
    assert "SPDX-FileCopyrightText: 2017 John Doe" in foo_file_info.copyright_lines
```

Explicit again to make the test more robust.